### ISSUE TITLE
[Bug]: clang: The output of UTF-8 messages when preprocessing diagnostics is not normal

### ISSUE THREAD

**Issue by dev-bz** [termux/termux-packages#9619]
It happens on `#error` and `#warning` messages, as does `clangd`

![IMG_20220329_113407.jpg](https://user-images.githubusercontent.com/32380878/160531631-57678171-d1ed-4225-be89-4693ba250e42.jpg)

```
echo '#warning 这是一个警告' | clang -E -P - -xc
```

**Comment by xtkoba** [comment]
Reproduces for me (Android 8.1.0 aarch64).

FWIW, I get the following on my GNU/Linux x86_64 environment, which should be the expected behavior:

```
$ echo '#warning 这是一个警告' | clang -E -P - -xc
<stdin>:1:2: warning:  [-W#warnings]
#warning 这是一个警告
 ^

1 warning generated.
```

that is, non-ASCII characters are suppressed in the `[-W#warnings]` line.

**Comment by thunder-coding** [comment]
I guess it got fixed in LLVM 14. Atleast for me on Android 11, aarch64. If anyone else can reproduce the issue with clang 14, feel free to comment the same.

**Comment by xtkoba** [comment]
Fix proposed upstream: https://reviews.llvm.org/D125620.

**Comment by dev-bz** [comment]
> Fix proposed upstream: https://reviews.llvm.org/D125620.

Why not keep all characters?

**Comment by xtkoba** [comment]
@dev-bz IDK

**Comment by xtkoba** [comment]
Fixed (upstream) in `clang` version `15.0.0`. Thanks.
